### PR TITLE
Fix mark partial causative

### LINKED ISSUE #3226: Mark variant as partial causative doesn't work
try with this one: https://scout-stage.scilifelab.se/cust000/WGS-prep-control-fam13-211129/ab52365c4cfae377cb2aae683e7f122c

Must be some bug introduced by [this old PR of mine](https://github.com/Clinical-Genomics/scout/pull/3069)

### PR DESCRIPTION
**Pull request by northwestwitch**
- Fix Partial causative assignment in cases with no OMIM or HPO terms (fix #3226)
- Partial causative OMIM select options in variant page (the ugly visualization of OMIM options is due to the change of how diagnoses are saved in case document)

<details>
<summary>Testing on cg-vm1 server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. Make sure the PR is pushed and available on [Docker Hub](https://hub.docker.com/repository/docker/clinicalgenomics/scout-server-stage)
1. `ssh <USER.NAME>@cg-vm1.scilifelab.se`
1. `sudo -iu hiseq.clinical`
1. `ssh localhost`
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `podman ps`
1. Stop the service with current deployed branch: `systemctl --user stop scout.target`
1. Start the scout service with the branch to test: `systemctl --user start scout@<this_branch>`
1. Make sure the branch is deployed: `systemctl --user status scout.target`
</details>

**How to test: locally**:
1. On main branch: try to assign a variant as partial causative. The OMIM select options will be super long if you have individual-level diagnoses in demo case
2. Switch to this branch and OMIM options should look fine
3. Switch back to main branch and remove all OMIM and HPO terms from the case: try to assign a variant as partial causative: confirm that you cant't
4. Switch to this branch and repeat: it should work instead

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
- [ ] code approved by
- [ ] tests executed by

### CODE REVIEW

File: scout/server/blueprints/variant/templates/variant/utils.html [hunk 1]
```diff
@@ -22,10 +22,7 @@ <h5 class="modal-title" id="causativeLabel">Mark variant as causative</h5>
                     <br>
                     <div>
                       <input type="checkbox" class="ios8-switch" id="partial_causative" name="partial_causative"
-                        data-toggle='collapse' data-target='#partial_phenotypes'
-                          {% if case.phenotype_terms|length == 0 and case.diagnosis_phenotypes|length == 0 %}
```
Comment (northwestwitch) [inline scout/server/blueprints/variant/templates/variant/utils.html:26]: This is the original problem of the bug (disabled checkbox to assign variant as partial causative)

File: scout/server/blueprints/cases/templates/cases/utils.html [hunk 2]
```diff
@@ -441,7 +441,7 @@ <h6><span class="fa fa-medkit ml-3 mt-2"></span>
                 <div class="col-8">
                   {% for omim in variant_phenotypes.omim_terms %}
                     <span class="badge badge-sm badge-secondary">
-                      <a class="text-white" target="_blank" href="http://omim.org/entry/{{omim}}">
+                      <a class="text-white" target="_blank" href="http://omim.org/entry/{{omim|replace('OMIM:','')}}">
```
Comment (dnil) [inline scout/server/blueprints/cases/templates/cases/utils.html:444]: 👍

**Review by dnil — APPROVED** [review]

File: scout/server/blueprints/variant/templates/variant/utils.html [hunk 3]
```diff
@@ -39,10 +36,16 @@ <h6><strong>Associated phenotype</strong></h6>
                   <!-- Phenotype could be specified using OMIM -->
```
[the remark this answers is not in the document]
Comment (dnil) [reply scout/server/blueprints/variant/templates/variant/utils.html:36]: Ok, will fix! Like this now:
Screenshot 2022-03-08 at 15 58 13

[the remark this answers is not in the document]
Comment (northwestwitch) [reply scout/server/blueprints/variant/templates/variant/utils.html:36]: Nice!

**Pull request merged**